src/Features/CSharp/Portable/CodeFixes/AddParameterXmlDocumentation/AddParameterXmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,316 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.AddParameterXmlDocumentation
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddParameterXmlDocumentaion), Shared]
+    internal partial class AddParameterXmlDocumentationCodeFixProvider : CodeFixProvider
+    {
+        // Parameter 'parameter' has no matching param tag in the XML comment for 'parameter' (but other parameters do)
+        private const string CS1573 = nameof(CS1573);
+        // Type parameter 'type parameter' has no matching typeparam tag in the XML comment on 'type' (but other type parameters do)
+        private const string CS1712 = nameof(CS1712);
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CS1573, CS1712);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        private BaseParameterListSyntax FindFirstAncestorParameterItem(SyntaxNode node)
+        {
+            for (; node != null; node = node.Parent)
+            {
+                if (node is BaseMethodDeclarationSyntax)
+                {
+                    return ((BaseMethodDeclarationSyntax)node).ParameterList;
+                }
+                if (node is IndexerDeclarationSyntax)
```

Comment:
Style: either use `else if` or add a blank line after the closing brace of each `if`